quickwit-indexing/src/actors/indexing_pipeline.rs

```diff
@@ -419,19 +421,44 @@ impl IndexingPipeline {
         Ok(())
     }
 
+    /// retry_count, wait_time_sec
+    /// 1   1
+    /// 2   4
+    /// 3   8
+    /// 4   16
+    fn wait_duration_before_retry(retry_count: usize) -> Option<Duration> {
+        let mut wait_time_sec = (retry_count + 1).pow(2);
+        wait_time_sec = wait_time_sec.min(60 * 5);
+        Some(Duration::from_secs(wait_time_sec as u64))
+    }
+
+    async fn process_spawn(
+        &mut self,
+        ctx: &ActorContext<IndexingPipelineMessage>,
+        retry_count: usize,
+    ) -> Result<(), ActorExitStatus> {
+        if let Err(spawn_error) = self.spawn_pipeline(ctx).await {
+            if let Some(duration_before_retry) = Self::wait_duration_before_retry(retry_count) {
+                error!(err=?spawn_error, "Error while spawn_pipeline, waiting 1000ms");
+                ctx.schedule_self_msg_blocking(
+                    duration_before_retry,
+                    IndexingPipelineMessage::Spawn {
+                        retry_count: retry_count + 1,
+                    },
+                );
+            } else {
+                return Err(ActorExitStatus::Failure(Arc::new(spawn_error)));
+            }
+        }
+
+        Ok(())
+    }
+
     async fn process_supervise(
         &mut self,
         ctx: &ActorContext<IndexingPipelineMessage>,
     ) -> Result<(), ActorExitStatus> {
-        if self.handlers.is_none() {
-            // TODO Accept errors in spawning. See #463.
-            self.spawn_pipeline(ctx).await?;
-            // if let Err(spawn_error) = self.spawn_pipeline(ctx).await {
-            //    // only retry n-times.
-            //    error!(err=?spawn_error, "Error while spawning");
-            //    self.terminate().await;
-            // }
-        } else {
+        if self.handlers.is_some() {
```

Comment:
I am lost... When do we respawn if workers have failed now?

Reply:
I added it to the observer now